coalib/io/FileFactory.py

```diff
@@ -0,0 +1,132 @@
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('name', 'timestamp')
+class FileFactory:
+    """
+    The ``FileFactory`` is a class to provide different views on a file.
+    The objects of this class contains the following information about the file:
+
+        * The filename (absolute file path).
+        * The modification timestamp of the file.
+        * The file in raw format.
+        * The file in the form of a string UTF-8 decoded from the raw format.
+        * The lines of the file in the form of a tuple.
+
+    Two ``FileFactory`` objects are conisdered equal if they have the
+    same filename and timestamp.
+
+    To initialize a ``FileFactory`` object for a file:
+
+    >>> import os
+    >>> import tempfile
+    >>> temp = tempfile.NamedTemporaryFile(delete=False)
+    >>> temp.write(bytes('This is a test file.', 'UTF-8'))
+    20
+    >>> temp.close()
+    >>> fp = FileFactory(temp.name)
+
+    File indices start with zero.
+    To retrieve a single line:
+
+    >>> fp.get_line(0)
+    'This is a test file.'
+
+    Get all the lines in the file as a list:
```

Comment:
But maybe say
> Get all the lines in the file at once:

Then you don't specify the actual return type, it's some object where you have access to all lines :) In our case it's a tuple to guarantee immutability, but that is an implementation detail to provide a reliable and safe API.